### PR TITLE
Add `CompositeEffectBuilder` to API

### PR DESCRIPTION
**Pull request by Forwindzz**
Hi,
Here is the code for
- `CompositeEffectBuilder` API
- Example cornerstone for `CompositeEffectBuilder`
- A tiny chinese translation fix in `ExampleMod`
 
This code may have some design problems:

(1) Function Name Problem:
In `CompositeEffectModel`, they have two fields:
`hasNestedPreview` -> preview like explode range
`hasNestedStatePreview` -> preview texts 
So I design API function names in this way:
`hasNestedPreview` -> preview like explode range -> API function name: `SetNestedPreviewActionIndex(int index)`
`hasNestedStatePreview` -> preview texts ->  API function name:`SetNestedPreviewIndex(int index)`
Otherwise, it is easy to confused with `HookedEffectModel.SetPreviewDescription()`
I wonder if the name `PreviewAction` is okay.

(2) Conflict Class Names:
`Eremite.Model.Effects.Hooked.TextArgType` (from `HookedEffectModel`)
`Eremite.Model.Effects.TextArgType` (from `CompositeEffectModel`)
They have the same class name...
So in the example mod, I use alias names.
```C#
using HookedTextArgType = Eremite.Model.Effects.Hooked.TextArgType;
using CompositeTextArgType = Eremite.Model.Effects.TextArgType; // note: these two TextArgType are different!
```
Is there any better design?

### CODE REVIEW

File: ATS_API/Scripts/Effects/EffectBuilders/CompositeEffectBuilder/CompositeEffectBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        m_effectModel.rewards = [];
+    }
+
+    /// <summary>
+    /// Add an effect to the composite effect
+    /// </summary>
+    /// <param name="effect"></param>
+    public void AddEffect(EffectModel effect)
+    {
+        m_effectModel.rewards.ForceAdd(effect);
```
Comment (JamesVeug) [inline ATS_API/Scripts/Effects/EffectBuilders/CompositeEffectBuilder/CompositeEffectBuilder.cs:35]: The API works off the assumption core EffectModels do not exist yet so mods can be written in the Awake method of their plugin.
Just realized HookedEffectBuilder also does this which isn't ideal...

Something to look into later is changing this method and others from changing m_effectModel directly but instead doing it in NewEffectData.PostSync and use EffectTypes as parameters instead.

File: ExampleMod/Plugin.Cornerstones.cs [hunk 2]
```diff
@@ -4,20 +4,23 @@
 using Eremite.Model.Effects;
 using Eremite.Model.Effects.Hooked;
 using UnityEngine;
-using TextArgType = Eremite.Model.Effects.Hooked.TextArgType;
+using HookedTextArgType = Eremite.Model.Effects.Hooked.TextArgType;
+using CompositeTextArgType = Eremite.Model.Effects.TextArgType; // note: these two TextArgType are different!
```
Comment (JamesVeug) [inline ExampleMod/Plugin.Cornerstones.cs:8]: Yeah this is awkward but no dunno what else we can do either!

File: ExampleMod/Plugin.Cornerstones.cs [hunk 3]
```diff
@@ -96,4 +103,28 @@ private void CreateGoodEveryYearProductionCornerstone()
         giveDiamondEffect.SetGood(1, diamonds.Name);
         DiamondHunterBuilder.AddHookedEffect(giveDiamondEffect.EffectModel);
     }
+
+    private void CreateAggregateProductivityBuilder()
+    {
+        AggregateProductivityBuilder = new CompositeEffectBuilder(PluginInfo.PLUGIN_GUID, "AggregateProductivity", "ModdingTools.png"); //Need to change the image
```
Comment (JamesVeug) [inline ExampleMod/Plugin.Cornerstones.cs]: Need new icon <3

**Review by JamesVeug — CHANGES_REQUESTED** [review]
Just requires an icon!

Note there that the API should be adding non-EffectModel support so people don't need to wait for the game to load to reference vanilla effects

Edit:
ElenaRoan and [Chaos] made the icons for the example mod so could ask them for assistance?

**Comment by Forwindzz** [comment]
I have added an icon

**Review by JamesVeug — APPROVED** [review]

**Pull request merged**